src/snowflake/snowpark/modin/plugin/extensions/base_overrides.py

```diff
@@ -1530,6 +1530,36 @@ def __getitem__(self, key):
     return self.loc[:, key]
 
 
+# Modin uses the unique() query compiler method instead of aliasing the duplicated frontend method as of 0.30.1.
+@register_base_override("drop_duplicates")
+def drop_duplicates(
+    self, keep="first", inplace=False, **kwargs
+):  # noqa: PR01, RT01, D200
+    """
+    Return `BasePandasDataset` with duplicate rows removed.
+    """
+    inplace = validate_bool_kwarg(inplace, "inplace")
+    ignore_index = kwargs.get("ignore_index", False)
+    subset = kwargs.get("subset", None)
+    if subset is not None:
+        if is_list_like(subset):
+            if not isinstance(subset, list):
+                subset = list(subset)  # pragma: no cover
+        else:
+            subset = [subset]
+        df = self[subset]
+    else:
+        df = self
+    duplicated = df.duplicated(keep=keep)
+    result = self[~duplicated]
+    if ignore_index:
+        result.index = pandas.RangeIndex(stop=len(result))
```

Comment:
where does this part of code coming from? is tht what we use in snowpark pandas before? it is wired that we use result.index for ignore_index, isn't that just a reset index?

Reply:
yes, I copied this code exactly as it was from 0.28.1: https://github.com/modin-project/modin/blob/eac21a8463d7152b3ad48a8b80d6da5ac645be6a/modin/pandas/base.py#L1503-L1528

I'm not sure why this index set is here intead of a reset_index either, but I just kept it as-is to avoid any potentially breaking changes

Reply:
Filed SNOW-1758721